platform/core/src/extensions/ExtensionManager.ts

```diff
@@ -109,6 +110,41 @@ export default class ExtensionManager extends PubSubService {
     );
   }
 
+  /**
+   * Calls all the services and extension on mode init.
+   * The service onModeInit is called first then registered extensions onModeInit is called.
+   * This is supposed to setup the extension for custom initial setup.
+   */
+  public onModeInit(): void {
```

Comment:
This is getting out of hand, can we refactor it into something like

```js
public onModeChange(mode: 'enter' | 'exit' |'init')
```

Reply:
I like your suggestion, it scales better. If that's what we want I'll go ahead and refactor it.